landscape/lib/tests/test_network.py

```diff
@@ -173,25 +173,32 @@ def test_skip_loopback(self):
         interfaces = [i["interface"] for i in device_info]
         self.assertNotIn("lo", interfaces)
 
-    @patch("landscape.lib.network.get_active_interfaces")
-    def test_skip_vlan(self, mock_get_active_interfaces):
+    @patch("landscape.lib.network.netifaces.interfaces")
+    def test_skip_vlan(self, mock_interfaces):
         """VLAN interfaces are not reported by L{get_active_device_info}."""
-        mock_get_active_interfaces.side_effect = lambda: (
-            list(get_active_interfaces()) + [("eth0.1", {})])
+        mock_interfaces.side_effect = lambda: (
+            _interfaces() + ["eth0.1"])
         device_info = get_active_device_info()
-        self.assertTrue(mock_get_active_interfaces.called)
         interfaces = [i["interface"] for i in device_info]
         self.assertNotIn("eth0.1", interfaces)
 
-    @patch("landscape.lib.network.get_active_interfaces")
-    def test_skip_alias(self, mock_get_active_interfaces):
+    @patch("landscape.lib.network.netifaces.interfaces")
+    def test_skip_alias(self, mock_interfaces):
         """Interface aliases are not reported by L{get_active_device_info}."""
-        mock_get_active_interfaces.side_effect = lambda: (
-            list(get_active_interfaces()) + [("eth0:foo", {})])
+        mock_interfaces.side_effect = lambda: (
+            _interfaces() + ["eth0:foo"])
         device_info = get_active_device_info()
         interfaces = [i["interface"] for i in device_info]
         self.assertNotIn("eth0:foo", interfaces)
 
+    @patch("landscape.lib.network.netifaces.ifaddresses")
+    @patch("landscape.lib.network.netifaces.interfaces")
+    def test_no_extra_netifaces_calls(self, mock_interfaces, mock_ifaddresses):
+        """Make sure filtered out interfaces aren't used in netiface calls"""
```

Comment:
"…due to their impact on sysinfo/login time with a large amount of interfaces."

It not obvious otherwise that this test is gating performance regression.